code/go/0chain.net/smartcontract/dbs/event/reward-delegate-test.go

```diff
@@ -0,0 +1,42 @@
+package event
+
+import "0chain.net/smartcontract/stakepool/spenum"
+
+func (edb *EventDb) GetRewardsToDelegates(blockNumber, startBlockNumber, endBlockNumber string, rewardType int) []RewardDelegate {
+
+	if blockNumber != "" {
+		var rds []RewardDelegate
+		edb.Get().Where("block_number = ? AND reward_type = ?", blockNumber, rewardType).Find(&rds)
+
+		return rds
+	}
+
+	if startBlockNumber != "" && endBlockNumber != "" {
+		var rds []RewardDelegate
+		edb.Get().Where("block_number >= ? AND block_number <= ? AND reward_type = ?", startBlockNumber, endBlockNumber, rewardType).Find(&rds)
+
+		return rds
+	}
+
+	return nil
```

Comment:
Please validate the input params, and return error for invalid params. We need to know what happens instead of return empty values to users.

Also, please also check the event db error.